core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -160,7 +166,7 @@ private void cacheChanges() {
     // read only manifests that were created by this snapshot
     Iterable<ManifestFile> changedManifests = Iterables.filter(manifests(),
         manifest -> Objects.equal(manifest.snapshotId(), snapshotId));
-    try (CloseableIterable<ManifestEntry> entries = new ManifestGroup(ops, changedManifests)
+    try (CloseableIterable<ManifestEntry> entries = new ManifestGroup(io, changedManifests, specsById)
```

Comment:
`specsById` is only needed when filtering the manifest group by partition fields. This is only using the entry content to filter. I don't think we need to pass the spec map here, in which case we also don't need to add it to `Snapshot` at all. That's much cleaner and allows us to remove a lot of these changes.

Reply:
Did you mean to pass one entry (a partition spec) instead the map?

(Sorry for the late response)

Reply:
My point is that I don't think we need to pass any specs into `ManifestGroup` for this path because we aren't adding partition filters. We are only filtering on entry metadata.

That means we can have a constructor for `ManifestGroup` that doesn't require a spec map, and that would allow us to avoid having the current spec map in `Snapshot`, which would simplify quite a bit.